frontend/src/metabase/setup/setup.unit.spec.tsx

```diff
@@ -0,0 +1,156 @@
+import { fireEvent, waitFor, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+import type { UsageReason } from "metabase-types/api";
+import {
+  createMockSettingsState,
+  createMockSetupState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { renderWithProviders, screen } from "__support__/ui";
+import { Setup } from "./components/Setup";
+
+const setup = async ({ step = 0 } = {}) => {
+  const state = createMockState({
+    setup: createMockSetupState({
+      step,
+    }),
+    settings: createMockSettingsState({
+      "available-locales": [["en", "English"]],
+    }),
+  });
+
+  fetchMock.post("path:/api/util/password_check", { valid: true });
+
+  renderWithProviders(<Setup />, { storeInitialState: state });
+
+  // there is some async stuff going on with the locale loading
+  await screen.findByText("Let's get started");
+
+  return;
+};
+
+describe("setup", () => {
+  describe("default step order", () => {
+    // eslint-disable-next-line jest/expect-expect
+    it.each([
+      ["What's your preferred language?", "1"],
+      ["What should we call you?", "2"],
+      ["What will you use Metabase for?", "3"],
+      ["Add your data", "4"],
+      ["Usage data preferences", "5"],
+    ])(
+      "should have a step for '%s' with label '%s'",
+      async (step: string, label: string) => {
+        await setup();
+        skipWelcomeScreen();
+
+        expectSectionToHaveLabel(step, label);
+      },
+    );
+  });
+
+  describe("Usage question", () => {
+    const setupForUsageQuestion = async () => {
+      await setup();
```

Comment:
The setup manually fills the previous steps, I know that mocking the redux state to directly load the correct step would be faster, but it would also give us less confidence and depend on some implementation details that will change with the refactor of the step navigation

Reply:
Done, thanks c0b86baa7dcaa595d91aca09a423a86050d69cb6